src/iosMain/kotlin/com/shopify/library/internal.star/IosStarSdk.kt

```diff
@@ -28,10 +28,17 @@ import platform.posix.u_int8_tVar
  */
 internal class IosStarSdk : StarSdk {
     override suspend fun searchPrinters(target: StarSdk.StarQuery): List<PortInfo> {
-        return SMPort.searchPrinter(target.query)
-            ?.filterIsInstance<StarIO.PortInfo>()
-            ?.map { IosPortInfo(it) }
-            ?: emptyList()
+        memScoped {
+            val searchPrintersError = alloc<ObjCObjectVar<NSError?>>()
+            val portInfos = SMPort.searchPrinter(target.query, searchPrintersError.ptr)
```

Comment:
Hilariously if you don't provide an error pointer, then the `searchPrinters()` function returns immediately with no results. If you _do_ provide the error pointer, it works like a charm 🤦